src/mc_rosgraph_manipulator/rosgraph_manipulator.py

```diff
@@ -104,17 +109,22 @@ def __init__(self, name, configFilePath):
     def callback(self, goal_msg):
         self.last_saved_goal = goal_msg.goal
 
+    def stop_configuration(self, req):
+        for node in self.kill_nodes:
+            kill_node(node)
+            rospy.loginfo("Stopping node %s" % str(node))
+            rospy.sleep(2)
+
+        return EmptyResponse()
+
     def reconfiguration_action_cb(self, goal):
         rospy.loginfo(
             'Rosgraph Manipulator Action Server received goal %s' % str(goal))
         self._result.result = 1
         desired_configuration = goal.desired_configuration_name
 
         if (desired_configuration in self.configurations):
-            for node in self.kill_nodes:
-                kill_node(node)
-                rospy.loginfo("Stopping node %s" % str(node))
-                rospy.sleep(2)
+            self.stop_configuration()
```

Comment:
wouldn't this fail because the `req` argument is missing?

maybe create a separate function for the service callback, and then just call the `stop_configuration()` so it's not conflicting